swift/Ice/helloUI/ProxySettings.swift

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+import Foundation
+
+class ProxySettings {
+    var connection = ""
+    var delay = 0.0
+    var timeout = 0.0
+    var deliveryMode: DeliveryMode = .twoway
+    var methodIndex = 0 {
+        didSet {
+            deliveryMode = updateDeliveryMode(index: methodIndex)
+        }
+    }
+
+    func updateDeliveryMode(index: Int) -> DeliveryMode {
+        return DeliveryMode(rawValue: index) ?? .twoway
+    }
+
+    var isBatched: Bool {
+        return DeliveryMode.batchedCases.contains(deliveryMode)
+    }
+
+    enum DeliveryMode: Int, CaseIterable {
```

Comment:
There's no point for this to have a raw type of `Int` anymore. You can _probably_ remove this.